distributed/tests/test_client.py

```diff
@@ -6461,19 +6461,28 @@ async def f(stacklevel, mode=None):
     assert "cdn.bokeh.org" in data
 
 
+@pytest.mark.skipif(
+    sys.version_info >= (3, 10), reason="No internal loop in Python 3.10"
```

Comment:
the "_loop" attribute is there in 3.10 - it's just late bound: https://github.com/python/cpython/blob/8acbb93c0763fa53b5959fe05d86ba275c9e8a5b/Lib/asyncio/mixins.py#L28

the tests could be modified to use the asyncio resource first so the loop is always bound before the assert, eg:

```python
await c.gather([c.submit(...)])
assert c._gather_semaphore._loop
```